lib/fastlane/setup/setup_ios.rb

```diff
@@ -87,36 +124,63 @@ def ask_to_enable_other_tools
         Helper.log.info 'Since all files are moved into the `fastlane` subfolder, you have to adapt your Deliverfile'.yellow
       else
         if agree("Do you want to setup 'deliver', which is used to upload app screenshots, app metadata and app updates to the App Store? This requires the app to be in the App Store already. (y/n)".yellow, true)
-          Helper.log.info "Loading up 'deliver', this might take a few seconds"
-          require 'deliver'
-          require 'deliver/setup'
-          options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {})
-          Deliver::Runner.new(options) # to login...
-          Deliver::Setup.new.run(options)
-
-          @tools[:deliver] = true
+          enable_deliver
         end
       end
 
       unless @tools[:snapshot]
         if Helper.mac? and agree("Do you want to setup 'snapshot', which will help you to automatically take screenshots of your iOS app in all languages/devices? (y/n)".yellow, true)
-          Helper.log.info "Loading up 'snapshot', this might take a few seconds"
-
-          require 'snapshot'
-          require 'snapshot/setup'
-          Snapshot::Setup.create(folder)
-
-          @tools[:snapshot] = true
+          enable_snapshot
         end
       end
 
-      @tools[:sigh] = true if agree("Do you want to use 'sigh', which will maintain and download the provisioning profile for your app? (y/n)".yellow, true)
+      if agree("Do you want to use 'sigh', which will maintain and download the provisioning profile for your app? (y/n)".yellow, true)
+        enable_sigh
+      end
+    end
+
+    def default_enable_other_tools
+      enable_deliver
+      # enable_snapshot
+      enable_sigh
+    end
+
+    def enable_sigh
+      @tools[:sigh] = true
+    end
+
+    def enable_snapshot
+      Helper.log.info "Loading up 'snapshot', this might take a few seconds"
+
+      require 'snapshot'
+      require 'snapshot/setup'
+      Snapshot::Setup.create(folder)
+
+      @tools[:snapshot] = true
```

Comment:
this is in an `enable_snapshot` method that gets called from the existing method `ask_to_enable_other_tools` i just abstracted it out of that method so that i could call it from another place also.